agent/util-scripts/tool-meister/pbench-sysinfo-dump

```diff
@@ -101,9 +105,19 @@ function collect_sos {
 		_modules="${_modules} tuned"
 	fi
 
+	if [[ "${sos_ver}" -ge 4 ]]; then
+		# the command is now "sos report" in 4.0+
+		_sos_report_cmd="sos report"
+	fi
+
+	if [[ "${sos_ver}" -gt 4 || ( "${sos_ver}" -eq 4 && "${sos_ver_minor}" -ge 1 )]]; then
+		# sos uses sha256 sums, not md5, in 4.1+
+		_cksum_suffix="sha256"
+	fi
+
 	_name="pbench-${_pbench_full_hostname}"
 	_cmd="${dir}/sosreport-${_name}.cmd"
-	printf -- "sosreport" > ${_cmd}
+	printf -- "%s" "${_sos_report_cmd}" > ${_cmd}
```

Comment:
While this obviously works, it's more complex than it needs to be; e.g.:
```
	printf -- "${_sos_report_cmd}" > ${_cmd}
```
If you're worried about there being percent signs in the command string, then use something even simpler, like:
```
	echo ${_sos_report_cmd} > ${_cmd}
```

Reply:
I am trying to make it a habit to never use a `printf` without a `%s` format spec when  printing anything but a fixed string. Feel free to whack me if I deviate from that :-)

Reply:
Fair enough.  Although, under the heading of "use the right tool for the job", I would suggest that, if you don't use `printf`, you won't run into that problem.  😉

Reply:
We can't use `echo` here because it adds a newline, and we are forming a command line in the file, `${_cmd}`.

So one could use `echo -n "${_cmd}"`.

I believe the quotes around the string are required to prevent the contents of the string from being interpreted by the shell.

Reply:
As you say, the `-n` switch resolves the newline problem.

I don't think the quotes are required:  the variable expansion is done during the shell's interpretation -- the contents don't get interpreted a second time.